.github/workflows/tests.yml

```diff
@@ -0,0 +1,38 @@
+name: Tests
+
+on:
+    pull_request:
+    push:
+        branches:
+        - master
+
+jobs:
+  build:
+    name: CI
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest]
+        resolver: [lts-16, lts-15, lts-14]
+
+    steps:
+      - name: Clone project
+        uses: actions/checkout@v2
+
+      - name: Cache dependencies
+        uses: actions/cache@v1
+        with:
+          path: ~/.stack
+          key: ${{ runner.os }}-${{ matrix.resolver }}-${{ hashFiles('stack.yaml') }}
```

Comment:
The cache depends on `xmonad-extras.cabal` as well, as adding dependencies results in more built packages.

Reply:
We could. But I don't want to invalidate the cache because the resolver is still the same and we can use other parts of the cache. 

But this current method will become inefficient if there is a huge amount of new packages newly  to `xmonad-extras.cabal`. But I don't think that will happen (I could be wrong). So the tradeoff here is cache getting invalidated often for some changes in the cabal file vs no cache available for some of the newly packages added in the cabal file. I took the second tradeoff, but I'm happy to be convinced otherwise.

Reply:
Ah, I see. Thanks, that does make sense. I will try to create a follow up PR unless someone beats me to it.